projects/demo/src/modules/app/version-manager/version-manager.providers.ts

```diff
@@ -10,11 +11,21 @@ export const SELECTED_VERSION_META = new InjectionToken<TuiVersionMeta | null>(
 export const VERSION_MANAGER_PROVIDERS: Provider[] = [
     {
         provide: SELECTED_VERSION_META,
-        deps: [LocationStrategy],
-        useFactory: (s: LocationStrategy): TuiVersionMeta | null => {
+        deps: [LocationStrategy, WINDOW],
+        useFactory: (
+            strategy: LocationStrategy,
+            window: Window,
+        ): TuiVersionMeta | null => {
+            const possibilityVersionSegment =
+                window.location.pathname.split(`/`)[1] ?? ``;
+
             return (
-                TAIGA_VERSIONS_META.find(
-                    meta => meta.baseHref === s.getBaseHref().replace(/\//g, ``),
+                TAIGA_VERSIONS_META.filter(
+                    meta =>
+                        possibilityVersionSegment === `` ||
+                        possibilityVersionSegment !== meta.baseHref,
```

Comment:
Can you explain me what happens in this `.filter(...)`, please ?
I dont understand the logic of filtering :(